packages/integrations/node/src/preview.ts

```diff
@@ -3,7 +3,11 @@ import http from 'http';
 import { fileURLToPath } from 'url';
 import { createServer } from './http-server.js';
 import type { createExports } from './server';
+import { polyfill } from '@astrojs/webapi';
 
+polyfill(globalThis, {
+	exclude: 'window document',
+});
```

Comment:
if you look at, for ex. the build, this is inside of the exported function, not a the top-level.